controller/user_controller.go

```diff
@@ -0,0 +1,56 @@
+package controller
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"log"
+	"net/http"
+
+	"github.com/fumist23/game-api/database"
+	"github.com/fumist23/game-api/model"
+)
+
+// /user/createに対するハンドラ
+// requestからnameを取り出してtoken生成してDBに保存して返す
+func CreateUser(w http.ResponseWriter, r *http.Request) {
+
+	body := r.Body
+	defer body.Close()
+
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
```suggestion
	ctx := r.Context()
```
context.Backgroundではなくてrequestからcontextは発行する方がいい
ここで空のを作ってしまうとサーバーとのコネクションが切れたタイミングとかでもキャンセルが走らない.
あとhttp.Request.context()でcontext呼ぶとライブラリ内でキャンセル処理入れてるからこちらで入れる必要がなくなる

Reply:
ここのやつか
https://golang.org/src/net/http/request.go?s=12287:12330#L323

Reply:
なるほどそうなんですね！🤔
学びです🙏

Reply:
https://github.com/golang/go/blob/44361140c02556a0a71bc52299149bb8de26024b/src/net/http/server.go#L1817
cancel処理はここ！